lib/state_machines/machine_collection.rb

```diff
@@ -27,7 +27,7 @@ def initialize_states(object, options = {}, attributes = {})
 
       each_value do |machine| 
         unless machine.dynamic_initial_state?
-          force = options[:static] == :force || !attributes.keys.include?(machine.attribute)
+          force = options[:static] == :force || !attributes.keys.map(&:to_sym).include?(machine.attribute)
```

Comment:
Actually this looks so messed up we should move this whole method to some other class aware of rails. I suspect something like this would be enough https://github.com/pluginaweek/state_machine/pull/333/files#diff-079a2f36f5788c9f5cc7b67e169cc497R26. what you think?